xlive/H2MOD/Modules/Shell/Startup/Startup.cpp

```diff
@@ -102,14 +103,16 @@ void InitLocalAppData()
 // by default useAppDataLocalPath is set to true, if not specified
 void prepareLogFileName(const wchar_t* logFileName, c_static_wchar_string<MAX_PATH>* path)
 {
-	const wchar_t* process_name = shell_is_dedicated_server() ? k_server_process_name : k_client_process_name;
+	const bool is_dedi = shell_is_dedicated_server();
+	const wchar_t* process_name = is_dedi ? k_server_process_name : k_client_process_name;
 	
 	// We set the instance string based on whether or not 
-	const wchar_t* instance_string;
+	const wchar_t* instance_string = NULL;
 	wchar_t instance_number_string[4];
-	if (config_use_instance_name())
+
+	if (config_use_instance_name(&instance_string))
 	{
-		instance_string =  g_shell_windows_instance_name;
+		// Do nothing, instance_string is set by the above function
```

Comment:
invert the if-else if no code is needed for if the condition is true

Reply:
changed